lib/File.php

```diff
@@ -39,10 +39,13 @@ class File
     public static function getFileExtension($name)
     {
         $name = strtolower($name);
-        $parts = explode('.', $name);
 
-        if (count($parts) > 1) {
-            return $parts[count($parts) - 1];
+        $pos = strrpos($name, '.');
+        if($pos) {
+            $extension = substr($name, $pos+1);
+            if ($extension && strpos($extension, '/') === false) {
+                return $extension;
+            }
```

Comment:
Why does not `pathinfo($path , PATHINFO_EXTENSION)` work here?

Reply:
I was not sure if `pathinfo()` works with all the possible (and currently used)  strings which are getting passed to this and I wanted to avoid any breaks here. We should change that in 10 I guess.